solo-chains/node/tanssi-relay-service/src/dev_service.rs

```diff
@@ -630,3 +945,44 @@ fn new_partial_basics(
         telemetry,
     })
 }
+
+use polkadot_primitives::{AvailabilityBitfield, UncheckedSigned, ValidatorId, ValidatorIndex};
+use sp_keystore::Error as KeystoreError;
+fn keystore_sign<H: Encode, Payload: Encode>(
+    keystore: &KeystorePtr,
+    payload: Payload,
+    context: &SigningContext<H>,
+    validator_index: ValidatorIndex,
+    key: &ValidatorId,
+) -> Result<Option<UncheckedSigned<Payload>>, KeystoreError> {
+    let data = payload_data(&payload, context);
+    let signature = keystore
+        .sr25519_sign(ValidatorId::ID, key.as_ref(), &data)?
+        .map(|sig| UncheckedSigned::new(payload, validator_index, sig.into()));
+    Ok(signature)
+}
+
+fn payload_data<H: Encode, Payload: Encode>(
+    payload: &Payload,
+    context: &SigningContext<H>,
+) -> Vec<u8> {
+    // equivalent to (`real_payload`, context).encode()
```

Comment:
Then why use the other way?

Reply:
I think in the end it does not really matter much, but I copied this from polkadot, so I guess let's maintain it this way

Reply:
ignore